src/components/Text/Text.tsx

```diff
@@ -1,37 +1,80 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React from 'react';
 import styles from './Text.module.css';
 
-export interface Props {
-  /**
-   * The rendered tag name of the Heading. The tag name should always relate to the [document outline](http://html5doctor.com/outlines/) and a tag name should never be chosen for its default aesthetic qualities. If a specific style is desired, use the `size` prop to manipulate the Heading style.
-   */
-  as: 'p' | 'span';
-  /**
-   * The child node(s)
-   */
-  children?: ReactNode;
+export type Size =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'body'
+  | 'sm'
+  | 'xs'
+  | 'caption'
+  | 'overline';
+
+export type Color =
+  | 'alert'
+  | 'base'
+  | 'brand'
+  | 'info'
+  | 'inherit'
+  | 'neutral'
+  | 'success'
+  | 'warning'
+  | 'white';
+
+export type Props = {
   /**
-   * CSS class names that can be appended to the component.
+   * Controls whether to render text inline (defaults to "p");
    */
+  as?: 'p' | 'span';
+  children: React.ReactNode;
```

Comment:
Stuck with using `children` to pass the content. Can be changed to `text` as per guidelines, but seems more natural to write
```
<Text>Lorem Ipsum</Text>
```
than
```
<Text text="Lorem Ipsum />
```
to mimic <p> or <span>, however can change.

Also easier to handle cases like [`TextColorInherit` story](https://github.com/chanzuckerberg/edu-design-system/pull/899/files#diff-5972c0a39278e3e54030442e0e738d4084181e92d647a0dcbd3ee36145e209faR130) using `children`

Reply:
@jinlee93 Yeah, for `Text` (and `Heading`, `TextPassage` and other text-specific components), `children` is the way to go. Supporting inner HTML tags (`<strong>`, `<sup>`, and so on) is a lot more elegant than using `dangerouslySetInnerHTML` on the component internals. 

This component has a specific usage (namely, very short text strings) so having `children` in place is manageable.

Reply:
Just flagging that it would be a good idea to have a session to go through some of the text-level components like `Text`, `Heading`, `TextPassage`, and some text-but-also-structural components like `PageHeader` and `Section`.

Reply:
Honestly it generally feels more natural to pass the content via `children` for most components :/

Reply:
> Just flagging that it would be a good idea to have a session to go through some of the text-level components like Text, Heading, TextPassage, and some text-but-also-structural components like PageHeader and Section.

Added a ticket: https://app.shortcut.com/czi-edu/story/189697/text-level-components

Reply:
Keeping children